### PR TITLE
refactor rust code

### PR DESCRIPTION
**Pull request by danieleades**

### CODE REVIEW

**Comment by danieleades** [comment]
marking as draft until #674 is merged

File: server/Cargo.toml [hunk 1]
```diff
@@ -29,3 +29,7 @@ tempfile = "3.8.1"
 
 [dev-dependencies]
 tempfile = "3.8.1"
+
+[lints.clippy]
+all = "deny"
+type_complexity = { level = "allow", priority = 1 }  # remove me
```
Comment (danieleades) [inline server/Cargo.toml:35]: this should be removed in future by someone familiar enough with the codebase to create descriptive type aliases (not me!)

**Comment by danieleades** [comment]
@sestinj any blockers to getting this merged?

**Comment by sestinj** [comment]
Thank you again for doing this! My Rust was...rusty

**Pull request merged**